src/components/TextWithMiddleEllipsis/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import type {StyleProp, TextStyle, ViewStyle} from 'react-native';
+import {View} from 'react-native';
+import Text from '@components/Text';
+import useThemeStyles from '@hooks/useThemeStyles';
+
+type TextWithMiddleEllipsisProps = {
+    /** The text to display */
+    text: string;
+
+    /** Additional styles */
+    style?: StyleProp<ViewStyle>;
+
+    /** Additional text styles */
+    textStyle?: StyleProp<TextStyle>;
+};
+
+function TextWithMiddleEllipsis({text, style, textStyle}: TextWithMiddleEllipsisProps) {
+    const styles = useThemeStyles();
+
+    const firstPart = text.substring(0, Math.ceil(text.length / 2));
+    const secondPart = text.substring(Math.ceil(text.length / 2));
```

Comment:
How do you make sure this won't cut off the last character of each part? 
I tested on Web and mWeb, with long file name, the last character of the first part is always cut off

<img width="364" alt="Screenshot 2025-04-09 at 10 09 50" src="https://github.com/user-attachments/assets/c37610ea-dc46-47be-ba2b-bc7a9ac2f634" />
<img width="431" alt="Screenshot 2025-04-09 at 10 12 44" src="https://github.com/user-attachments/assets/c62392e6-4984-4031-b2a3-4acada97f2cb" />


<img width="1090" alt="Screenshot 2025-04-09 at 10 09 18" src="https://github.com/user-attachments/assets/05c73098-a6ae-470c-aac9-a564655df0e5" />

Reply:
Yeah, that's the problem I've faced on the web with text elipsis

I see that on your screenshot, It looks even worse than I've faced

I will check more with the ref approach on the first part then

Reply:
Is it critical for us?

 I've tried different approaches - with refs and element creation to measure. Hidden element creation to measure and cut the string looks more stable, but we will have computations for every element. (It still needs more time to build a reliable solution with it.)

Reply:
Not critical at the moment but I believe we still need to resolve the cut off because it's easy to notice

Reply:
Do you have any thoughts about handling this case with less complexity?

Reply:
I'm thinking about a workaround, the first text with elipse mode tail and make sure the second text wont be cut off.

<Text ellipseSize="tail">The first verylong text</Text> <Text>Thesecond text with no ellipse size</Text>

What do you think?

Reply:
I see the hardest part here, above solution doesn't work. Let me see if I can find any

Reply:
Just found a dude on GitHub using canvas approach (https://github.com/zhiyuc123/react-truncate-inside/blob/main/src/index.tsx).

Here's the demo:
https://zhiyuc123.github.io/react-truncate-inside/?path=/docs/demo--docs

I tried to implement it in our App and it looks well, can you take a look?

Reply:
Did you try it with middle truncation? 
 
 As I see this component using a predefined width and truncation by the number of symbols 
 
 I will try this, thanks

Reply:
I've checked, looks like this approach is applicable here 

I'm experimenting with a hidden div for now, since it has better measurements